qcodes/tests/dataset/test_data_set_cache.py

```diff
@@ -122,19 +128,29 @@ def test_cache_1d(experiment, DAC, DMM, n_points, bg_writing,
                                                     shaped_partial=set_shape)
     _assert_parameter_data_is_identical(dataset1.get_parameter_data(),
                                         dataset1.cache.data())
-    assert dataset1.cache._loaded_from_completed_ds is True
+    if in_memory_cache is False:
+        assert dataset1.cache._loaded_from_completed_ds is True
+    else:
+        # todo enable live cache to set this
```

Comment:
this comment can go, right?

Reply:
I would like to eventually find a way for the live cache to mark the dataset as completed

Reply:
oh, hmmm. why would we want that? at the first sight seems to me like increase of coupling between dataset and its cache and i don't see what this brings. could you share more?

Reply:
Mostly I would like that the dataset could be marked completed even if not written to disk